servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -278,13 +278,17 @@ public void removeBlocks(List<Long> blockIds) {
           if (masterBlockInfo == null) {
             continue;
           }
+          ArrayList<Long> workersToRemove = new ArrayList<Long>();
           for (long workerId : masterBlockInfo.getWorkers()) {
-            masterBlockInfo.removeWorker(workerId);
+            workersToRemove.add(workerId);
```

Comment:
What about this: for(long workerId : new ArrayList(masterBlockInfo.getWorkers()))?